### ISSUE TITLE
Specify the full path to clang in our toolchain setup [phab:D1146]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1508]
_Created by **salguarnieri** at 2016-11-11 17:44:42_: 

On OS X, we were not specifying the full path to clang, which meant that
sometimes it would not be properly copied into the sandbox and builds
would fail because they could not find clang. See the resolved ticket
#1500 for more details on
the problem and the fix.

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-11-11 17:49:36_:

This actually needs to point to the 'clang' executable in the repository (generated from the `osx_gcc_wrapper.sh.in` template, not the compiler which is found in the path).

**Comment by kythe** [comment]
_**shahms** wrote at 2016-11-11 17:56:42_:

It might actually be worth trying to find out what's necessary to make the upstream rules usable by us (off the top of my head, using the 'CLANG' environment variable and not looking for 'gcc' at all, as we are not compatible therewith).

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-11-11 18:04:02_:

It appears as though this part is actually correct, but we need to specify the dependency in the cc_toolchain() rule.

**Comment by kythe** [comment]
_**salguarnieri** wrote at 2016-11-11 19:53:58_:

  - add dep

**Comment by kythe** [comment]
_**salguarnieri** commented inline (context lost during migration) at 2016-11-11 20:05:04_:

I'm not sure how to do that, but I gave it a shot in the new patch set.

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-11-11 20:42:14_:

Based on my reading of the upstream equivalent, I think you want:
```
filegroup(
  name = "dawrin_files",
  srcs = ["clang", "clang++", ":empty"]
)

...

cc_toolchain(
  ...
  all_files = ":darwin_files",
  linker_files = ":darwin_files",
  compiler_files = ":darwin_files",
  ...
)
```

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-11-11 20:42:14_:

I *think* this should still just be "clang" (while applying the other changes to the BUILD file) but I'm not 100% on that.

**Comment by kythe** [comment]
_**salguarnieri** wrote at 2016-11-11 22:18:13_:

  - add dep
  - rename wrapper script to clang_wrapper_is_not_clang and fix up deps

**Comment by kythe** [comment]
_**salguarnieri** wrote at 2016-11-11 22:20:00_:

  - remove comment

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-11-11 22:21:41_:

Delete comment?

**Comment by kythe** [comment]
_**shahms** wrote at 2016-11-11 22:21:41_:

Thanks!

**Comment by kythe** [comment]
_**shahms** accepted this revision at 2016-11-11 22:22:30_

**Comment by kythe** [comment]
_**salguarnieri** closed via commit 4b81cb0801716a42a56c8d5a28a6ec0bbb2f9bd5 at 2016-11-11 22:27:31_
